sds/src/scanner/mod.rs

```diff
@@ -1237,6 +1258,27 @@ mod test {
             matches[0].path,
             Path::from(vec![PathSegment::Field("a-match".into())])
         );
+
+        let snapshot = snapshotter.snapshot().into_vec();
+        assert_eq!(snapshot.len(), 3);
+
+        let metric_name = "false_positive.multipass.excluded_match";
+        let metric_pos = snapshot
```

Comment:
I would recommend converting the `snapshot` [into a HashMap](https://docs.rs/metrics-util/0.16.3/metrics_util/debugging/struct.Snapshot.html#method.into_hashmap) so you can directly lookup by key.